executor/join.go

```diff
@@ -889,12 +911,44 @@ func (e *NestedLoopApplyExec) Next(ctx context.Context, req *chunk.Chunk) (err e
 			e.hasMatch = false
 			e.hasNull = false
 
-			for _, col := range e.outerSchema {
-				*col.Data = e.outerRow.GetDatum(col.Index, col.RetType)
-			}
-			err = e.fetchAllInners(ctx)
-			if err != nil {
-				return err
+			if e.canUseCache {
+				var key []byte
+				for _, col := range e.outerSchema {
+					*col.Data = e.outerRow.GetDatum(col.Index, col.RetType)
+					key, err = codec.EncodeKey(e.ctx.GetSessionVars().StmtCtx, key, *col.Data)
```

Comment:
Can we use HashChunkRow?

Reply:
Yes, we can, but the time cost for encoding these keys is negligible in `Apply`, and using `EncodeKey` here makes it simple and easy to understand. So how about keeping it?